test/e2e/main_test.go

```diff
@@ -0,0 +1,56 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"reflect"
+	"runtime"
+	"testing"
+
+	"github.com/knative/eventing/test"
+)
+
+// channelTestMap indicates which test cases we want to run for a given CCP.
+var channelTestMap = map[string][]func(t *testing.T){
```

Comment:
As mentioned in  person, I think this would be clearer as map[test]provisioners. However, that can be handled in a follow up PR.

Reply:
This is one of the options we have, I'll collect all the ideas and draft a design doc for further discussion.